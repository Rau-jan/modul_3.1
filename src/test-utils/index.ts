import { render as originRender } from '@testing-library/react';
import userEvent, { type UserEvent } from '@testing-library/user-event';

// Re-export
export * from 'vitest';

// Re-export
export * from '@testing-library/react';

type RenderArgs = Parameters<typeof originRender>;
type RenderResult = ReturnType<typeof originRender> & { user: UserEvent };
type RenderFn = (...args: RenderArgs) => RenderResult;

export const render: RenderFn = (elem, options) => {
  const user = userEvent.setup();
  const result = originRender(elem, options);
  return { user, ...result };
};
