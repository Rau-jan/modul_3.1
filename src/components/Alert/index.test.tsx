import { Alert } from '.';
import { render, expect, test } from '@/test-utils';

test('show component', () => {
  const { container } = render(
    <Alert type="warning">
      <div>Test</div>
    </Alert>,
  );
  expect(container).matchSnapshot();
});
