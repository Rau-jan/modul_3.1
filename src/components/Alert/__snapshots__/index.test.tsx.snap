// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`show component 1`] = `
<div>
  <div
    class="alert alert-dismissible fade show mb-0 alert-warning"
    role="alert"
  >
    <div>
      Test
    </div>
    <button
      aria-label="Close"
      class="btn-close"
      data-bs-dismiss="alert"
      type="button"
    />
  </div>
</div>
`;
